shapely/strtree.py

```diff
@@ -209,43 +272,161 @@ def query(self, geom: BaseGeometry) -> Sequence[BaseGeometry]:
         """
         return self.query_geoms(geom)
 
-    def _nearest_idx(self, geom: BaseGeometry, exclusive: bool = False) -> int:
+    def query_bulk(self, geometry, predicate=None, distance=None):
+        """Returns all combinations of each input geometry and geometries in the tree
+        where the envelope of each input geometry intersects with the envelope of a
+        tree geometry.
```

Comment:
I think query_combinations may be more descriptive than query_bulk?

Reply:
I don't have a strong opinion on the exact name (cc @brendan-ward who I think designed this initially)

The reason for the "bulk" is because it is basically a version of `query` that does the query for an array of geometries at once ("in bulk"). 
(in the end, the `query` method also returns the combinations but only with a single input geometry)

Reply:
I'm not at all attached to `_bulk` and never particularly liked the term; we came to that point to try to differentiate it from the `_all` functions in pygeos which behaved differently (though `nearest_all` is yet again different).

`_combinations` doesn't seem to add clarity, since both methods involve combinations (1 vs N tree geoms for `query`,  M vs N geoms for `query_bulk`).

I think what we're getting at here is that this is the vectorized form of query (input is a vector instead of a scalar), but also that it isn't 1:1 vectorized to the inputs (where 1:1 vectorized returns same shape as input).  So, `query_vec` or somesuch may work?  Something to think about for some of the other functions that clearly need to differentiate themselves as only taking vector inputs (at least for the primary input param).

Reply:
Given that the differentiation is also that the one takes a single geometry as input, and the "bulk" version an array of geometries, another option for a name could be `query_array`. 

Although, that also doesn't fully mach with the `query_geoms` and `query_items` terminology where the word after the `_` is what is being returned, not what is passed as input.

Reply:
Sorry, I don't have strong opinions on the name for this.  Names are hard!